vignettes/advs.Rmd

```diff
@@ -0,0 +1,313 @@
+
+The only difference here would be the the additional parameters collected in 
+metabolic trials - for example:
+
+`VSTESTCD` | `PARAMCD` | `PARAM` | `PARAMN` | `PARCAT1` | `PARCAT1N`
+--------- | --------- | -------- | ------- | --------- | ----------
+BMI | BMI | Body Mass Index (kg/m^2) | 1 | Subject Characteristic | 1
+HIPCIR | HIPCIR | Hip Circumference (cm) | 3 | Subject Characteristic | 1
+WSTCIR | WSTCIR | Waist Circumference (cm) | 5 | Subject Characteristic | 1
+
+```{r, message=FALSE, include=FALSE}
+library(admiral)
+library(admiralmetabolic)
+library(pharmaversesdtm)
+library(dplyr, warn.conflicts = FALSE)
+library(lubridate)
+library(stringr)
+
+data("dm_metabolic")
+data("vs_metabolic")
+
+dm <- convert_blanks_to_na(dm_metabolic)
+vs <- convert_blanks_to_na(vs_metabolic)
+
+data("admiral_adsl")
+
+dm <- dm %>%
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+
+adsl <- dm %>%
+  select(-DOMAIN)
+
+adsl <- adsl %>%
+  mutate(
+    TRT01P = ARM,
+    TRT01A = ACTARM
+  )
+
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+advs <- derive_vars_merged(
+  vs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = exprs(STUDYID, USUBJID)
+)
+
+advs <- derive_vars_dt(advs, new_vars_prefix = "A", dtc = VSDTC)
+
+advs <-
+  derive_vars_dy(advs, reference_date = TRTSDT, source_vars = exprs(ADT))
+
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "HEIGHT", "HEIGHT", "Height (cm)", 2,
+  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 3,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 4,
+  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+
+advs <- derive_vars_merged_lookup(
+  advs,
+  dataset_add = param_lookup,
+  new_vars = exprs(PARAMCD),
+  by_vars = exprs(VSTESTCD)
+)
+
+advs <- mutate(
+  advs,
+  AVAL = VSSTRESN
+)
+```
+
+## Derive Additional Parameters (e.g. `BMI` for `ADVS`) {#derive_param}
+
+In metabolic trials, `BMI` is often calculated at source using collected height and weight for screening purposes. But while creating the `ADVS` dataset, we re-derive `BMI` from the collected height and weight values. This is done to ensure that the `BMI` is calculated consistently across all subjects and visits. 
+
+In this step, we create parameter `Body Mass Index` (BMI) for `ADVS` domain using the `derive_param_bmi()` function.
+Note that only variables specified in the `by_vars` argument will be populated in the newly 
+created records. Also note that if height is collected only once use `constant_by_vars` to specify the subject-level variable to merge on.  Otherwise BMI is only calculated for visits where both are collected.
+
+
+```{r eval=TRUE}
+# Removing BMI collected at source from the dataset
+advs <- advs %>% filter(!VSTESTCD == "BMI")
+
+advs <- derive_param_bmi(
+  advs,
+  by_vars = exprs(STUDYID, USUBJID, !!!adsl_vars, VISIT, VISITNUM, ADT, ADY, VSTPT, VSTPTNUM),
```

Comment:
Use `get_admiral_option("subject_keys")` rather than explicitly naming STUDYID and USUBJID - see how it's done in [this vignette](https://pharmaverse.github.io/admiralophtha/articles/advfq.html) for an example